builtin/refs.c

```diff
@@ -101,6 +107,107 @@ static int cmd_refs_verify(int argc, const char **argv, const char *prefix,
 	return ret;
 }
 
+struct list_options {
+	int show_head;
+	int filter_branches;
+	int filter_tags;
+	int found_match;
+	const char **patterns;
+};
+
+static void print_ref(const char *refname, const struct object_id *oid)
+{
+	const char *hex;
+
+	if (!has_object(the_repository, oid,
+			HAS_OBJECT_RECHECK_PACKED | HAS_OBJECT_FETCH_PROMISOR))
+		die("git refs list: bad ref %s (%s)", refname,
+		    oid_to_hex(oid));
```

Comment:
This was unexpected, seems like git-show-ref(1) also does object verification. This can really slow down the command, do we want this? git-for-each-ref(1) doesn't do this AFAIK.

Reply:
Due to `git-show-ref(1)` does object verification, I suggested Meet to do above in the first version. I think we need to think about this design question: whether we need to do check.